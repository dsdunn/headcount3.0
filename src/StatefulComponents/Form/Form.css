form {
  margin: 5%;
  text-align: center;
}

h1 {
  font-size: 2.6em;
}

input {
  font-size: 1.5em;
  width:250px;
}

@media only screen and (max-width: 350px) {
  input {
    height: 5vh;
    font-size: 2em;
  }
}