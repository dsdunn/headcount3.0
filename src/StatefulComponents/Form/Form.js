import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  handleChange = (e) => {
    const value = e.target.value;
    this.setState({
      value
    });
    this.props.findFromSearch(value);

  }

  handleSubmit = (event) => {
    event.preventDefault();
    if (!event.target.value) {
      return;
    }
    this.props.selectDistrict(this.state.value);
    this.setState({value: ''});
    this.props.findFromSearch('');

  }

  render(){
    return (
      <form onSubmit={this.handleSubmit}>
        <h1>Head Count Logo</h1>
        <input
          className='searchField' 
          type="text"
          value={this.state.value}
          placeholder='Enter District'
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

export default Form;