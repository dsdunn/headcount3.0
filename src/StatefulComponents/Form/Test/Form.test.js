import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Form from '../Form';

describe('', ()=>{
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Form findFromSearch={jest.fn()}/>);

  });

  it('should match snapshot ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should instantiate with some state', () => {
    //setup
    const mockState = {
      value: ''
    };
    //execution
    //expectation
    expect(wrapper.instance().state).toEqual(mockState);
  });

  it('should call handleChange function when input changes', () => {
    //setup
    //mock function
    const mockHandleChange = jest.fn();
    const mockEvent = {target: {value: 'turing'}}
    //set state of wrapper
    
    //simulate change
    wrapper.find('.searchField').simulate('change', mockEvent);
    //execution
    //expectation
    expect(wrapper.instance().handleChange).toHaveBeenCalled;
    expect(wrapper.state('value')).toEqual('turing');
  });
});