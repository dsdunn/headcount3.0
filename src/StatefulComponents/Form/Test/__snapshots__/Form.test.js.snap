// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    findFromSearch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Head Count Logo
</h1>,
        <input
          className="searchField"
          onChange={[Function]}
          placeholder="Enter District"
          type="text"
          value=""
/>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Head Count Logo",
        },
        "ref": null,
        "rendered": "Head Count Logo",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchField",
          "onChange": [Function],
          "placeholder": "Enter District",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Head Count Logo
</h1>,
          <input
            className="searchField"
            onChange={[Function]}
            placeholder="Enter District"
            type="text"
            value=""
/>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Head Count Logo",
          },
          "ref": null,
          "rendered": "Head Count Logo",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchField",
            "onChange": [Function],
            "placeholder": "Enter District",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
